generate/base.py

```diff
@@ -142,13 +142,20 @@ def main(
     fabric.print(f"Time to instantiate model: {time.perf_counter() - t0:.02f} seconds.", file=sys.stderr)
 
     t0 = time.perf_counter()
-    with lazy_load(checkpoint_path) as checkpoint:
-        model.load_state_dict(checkpoint.get("model", checkpoint), strict=quantize is None)
-    fabric.print(f"Time to load the model weights: {time.perf_counter() - t0:.02f} seconds.", file=sys.stderr)
+
+    if quantize:
+        with lazy_load(checkpoint_path) as checkpoint:
+            # for quantization, need to load before moving to device
+            model.load_state_dict(checkpoint.get("model", checkpoint), strict=False)
 
     model.eval()
     model = fabric.setup_module(model)
 
+    if not quantize:
+        fabric.load_raw(checkpoint_path, model, strict=True)
```

Comment:
Could you use the loading utility that you added in #564?

Reply:
I need to rewrite it or provide a different utility then, because of this funny business 😅  :

```
checkpoint.get("model", checkpoint)
```